build/dsc.rake

```diff
@@ -80,6 +80,24 @@ eod
         Rake::Task['dsc:resources:checkout'].invoke(
           official_dsc_resources_root, update_versions, composite_resources)
 
+        # In PR https://github.com/PowerShell/SqlServerDsc/issues/851 the xSQLServer DSC Resource was
+        # renamed from xSQLServer to SqlServerDsc. This normally wouldn't be a problem, when they changed
+        # the name from, they updated the xSQLServer submodule to point to the commit in the newly
+        # renamed SqlServerDsc repo. Since we do a direct git submodule update, it uses the folder
+        # name of the submodule reference (xSQLServer) and not the name of the target repo (SqlServerDSC),
+        # resulting in a folder called xSQLServer with all the contents of the SqlServerDsc repo.
+        # The simple short term fix is to move in place the folder to the correct name. The longer term fix
+        # is stop using the powershell/dscresources git repo as a source of truth, which is part of a larger
+        # project that planned.
+        # This is hardcoded and done in a single line right now because premature optimization
+        # is the root of all evil. While we can predict there will be more renamed
```

Comment:
My PR comment at https://github.com/puppetlabs/puppetlabs-dsc/pull/339#discussion_r166495208 was just about maintenability - i.e. a small well-named method that would be called like this:

```ruby
resource_renames = [{:from => 'xSQLServer', :to => 'SqlServerDsc'}]
rename_resources(resource_renames)
```

Premature optimization is historically in reference to performance, so that doesn't apply here. YAGNI is closer (the Fowler def at https://martinfowler.com/bliki/Yagni.html is good), but doesn't really apply either since YAGNI is more about unused features than organizing code for maintainability.

The first few lines of the explanation about why xSQLServer requires the disk shuffling is good, but much of the rest of the commentary doesn't seem necessary.